onnxruntime/test/util/include/test_utils.h

```diff
@@ -20,17 +20,20 @@ namespace test {
 // If set to None: verify that no nodes is assigned to ep (typically for an expected failure path test case)
 enum class ExpectedEPNodeAssignment { None,
                                       Some,
-                                      All, };
+                                      All,
+};
 
 // struct to hold some verification params for RunAndVerifyOutputsWithEP
 struct EPVerificationParams {
-  
   ExpectedEPNodeAssignment ep_node_assignment = ExpectedEPNodeAssignment::Some;
 
   // Some EP may use different rounding than ORT CPU EP, which may cause a bigger abs error than
   // the default of 1e-5f, especially for scenarios such as [Q -> Quantized op -> DQ]
   // Set this only if this is necessary
   float fp32_abs_err = 1e-5f;
+
+  // optional graph verification function
+  const std::function<void(const Graph&)>* graph_verifier{nullptr};
```

Comment:
>const std::function<void(const Graph&)>* [](http://example.com/codeflow?start=2&length=40)

nit: could just be std::function, a default constructed function is empty.
to call it: `if (graph_verifier) graph_verifier(graph);`
[https://en.cppreference.com/w/cpp/utility/functional/function/operator_bool](https://en.cppreference.com/w/cpp/utility/functional/function/operator_bool)

Reply:
std::optional is a good signal too :)
perhaps a pointer is more obvious - I don't have a strong opinion here